src/efi/efisupport.c

```diff
@@ -110,11 +119,75 @@ static int find_free_boot_rec(void) {
     return res;
 }
 
-int efi_create_boot_rec(void) {
+typedef struct part_info {
+    char *disk_path;
+    int part_no;
+    char *part_type;
+} part_info_t;
+
+/* Given the location of the booloader, returns the partition information needed
+ * to create boot variable which points to that bootloader. */
+int get_part_info(const char *path, part_info_t *pi) {
```

Comment:
I wonder if Ikey might have some ideas on reusing existing code that does this kind of thing already. I'm not really familiar with this part of cbm though.